src/services/sets.tsx

```diff
@@ -38,5 +38,40 @@ const getIdsFromSaveSetId = async (rawId: string) => {
   return firstEdge?.node?.ids;
 };
 
+export const saveSetCountForTag = async (tag: string, userId: string) => {
+  const response = await graphql(initializeApi())({
+    query: `query($sqon: JSON) {
+        sets {
+          aggregations(filters: $sqon, aggregations_filter_themselves: false) {
+            size {
+              stats {
+                count
+              }
+            }
+          }
+        }
+      }`,
+    variables: {
+      sqon: {
+        op: 'and',
+        content: [
+          {
+            op: 'in',
+            content: { field: 'tag.keyword', value: [tag] },
+          },
+          {
+            op: 'in',
+            content: { field: 'userId', value: [userId] },
+          },
+        ],
+      },
+    },
+  });
+  return response?.data?.sets?.aggregations?.size?.stats?.count || 0;
```

Comment:
why all these checks? Wouldn't it be better to throw an error is something that is not expected occurs?